web/packages/teleterm/src/services/pty/ptyService.ts

```diff
@@ -19,38 +19,44 @@
 import { ChannelCredentials } from '@grpc/grpc-js';
 
 import { RuntimeSettings } from 'teleterm/mainProcess/types';
+import { ConfigService } from 'teleterm/services/config';
 
 import { buildPtyOptions } from './ptyHost/buildPtyOptions';
 import { createPtyHostClient } from './ptyHost/ptyHostClient';
 import { createPtyProcess } from './ptyHost/ptyProcess';
-import { PtyServiceClient, PtyOptions } from './types';
+import { PtyServiceClient } from './types';
 import { getWindowsPty } from './ptyHost/windowsPty';
 
 export function createPtyService(
   address: string,
   credentials: ChannelCredentials,
   runtimeSettings: RuntimeSettings,
-  options: PtyOptions
+  configService: ConfigService
 ): PtyServiceClient {
   const ptyHostClient = createPtyHostClient(address, credentials);
 
   return {
     createPtyProcess: async command => {
-      const windowsPty = getWindowsPty(runtimeSettings, options.terminal);
-      const { processOptions, creationStatus } = await buildPtyOptions(
-        runtimeSettings,
-        {
-          ssh: options.ssh,
-          windowsPty,
-        },
-        command
-      );
+      const windowsPty = getWindowsPty(runtimeSettings, {
+        windowsBackend: configService.get('terminal.windowsBackend').value,
+      });
+      const { processOptions, creationStatus, openedShell } =
```

Comment:
Ah, I see where `openedShell` came from. Still, I think `buildPtyOptions` should just return `shell` instead, it doesn't open any shell itself.